community/3-contributor.md

```diff
@@ -1,107 +1,108 @@
+[apache/incubator-shenyu-nginx](https://github.com/apache/incubator-shenyu-nginx)
 
-- Send a pull request to the master branch.
-- The mentor will do code review before discussing some details (including the design, the implementation and the performance) with you. The request will be merged into the branch of current development version after the edit is well enough.
-- At last, congratulate to be an official contributor of ShenYu.
+### Getting Apache ShenYu up and running
 
-**Delete Branch**
+To get Apache ShenYu code running on your development tools, see [Local Deployment](../docs/deployment/deployment-local) or [Apache ShenYu Startup Example](../blog/Start-SourceCode-Analysis-Start-Demo), and able to debug with breakpoints.
 
-- You can delete the remote branch (origin/issueNo) and the local branch (issueNo) associated with the remote branch (origin/issueNo) after the mentor merged the pull request into the master branch of ShenYu.
+After running, you can read the [source code analysis article](../blog) included in our official website to have a deeper understanding of Apache ShenYu.
 
-```shell
-git checkout master
-git branch -d issueNo
-git push origin --delete issueNo
-```
+### Find tasks
+
+Find the issue you are interested in! On our GitHub repo and mailing list, we often publish some issues with the label `good first issue` or `status: volunteer wanted`. These issues welcome the help of contributors. Among them, good first issues tend to have low thresholds and are suitable for novices. You can click the [link](https://github.com/apache/incubator-shenyu/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22%2C%22status%3A+volunteer+wanted%22) to check out these issues.
+
+Of course, if you have a good idea, you can also propose it directly on the mailing list, and after a thorough discussion, you can start to act.
+
+At the same time, you can also participate in the writing of our source code analysis articles and submit them to the blog module of the Apache ShenYu official website through Pull Request.
+
+If you are a student, you are also very welcome to apply for the topic of Apache ShenYu in open source internship activities such as GSoC. You can click the [link](https://community.apache.org/gsoc.html) to view the Apache Software Foundation Introduction to GSoC. You can also check Apache ShenYu's previous or ongoing GSoC issues through this [link](https://github.com/apache/incubator-shenyu/issues?q=is%3Aopen+is%3Aissue+label%3Agsoc)!
+
+### Submit Pull Request
+
+First you need to fork your target repository.
 
-**Notice**:  Please note that in order to show your id in the contributor list, don't forget the configurations below:
+Then, you can refer to the following command to submit the code:
+
+Switch to new a development branch
 
 ```shell
-git config --global user.name "username"
-git config --global user.email "username@mail.com"
-```
+git checkout -b a-new-branch
+````
+
+Do a commit
 
-**FAQ**
+```shell
+git add .
```

Comment:
may be   `git add <modify file path>` is better ?

Reply:
Yep, +1